### DISCUSSION TITLE
Can someone help me to do a 1G1R of FB Neo and/or MAME?

### DISCUSSION THREAD

**Discussion by notNSANE** [emmercm/igir#1344] [Q&A]
I'm not great on CLIs and I honestly don't trust RomCenter for this job. What is the command line I should use if I want to have this:

* US region, English Language for everything
*  * If not available, World or EU (English if possible) then JP (English if possible)

It's as simple as that. Any help appreciated! I'm going to use RetroArch so I believe I can throw the BIOSes into their /system/ directory.

**Comment by emmercm** [comment]
You'll need to figure out some additional information first per [the docs](https://igir.io/usage/arcade/):

- The correct DAT for the exact version of your emulator (you'll need to pick one)
- What type of "merge" would you like

My [personal usage](https://igir.io/usage/personal/#primary-rom-library) looks like this:

```shell
npx --yes igir@latest move zip test clean \
  `# Official MAME XML extracted from the progetto-SNAPS archive` \
  --dat "./mame*.xml" \
  --input "./MAME/" \
  --input-checksum-quick \
  --input-checksum-archives never \
  --output "./MAME/" \
  --dir-dat-name \
  --overwrite-invalid \
  --merge-roms merged \
  -v
```

**Comment by notNSANE** [comment]
what's npx? haha.

after some fiddling I got this but I don't know if it is overkill: 

`igir.exe copy --dat "FinalBurn Neo (ClrMame Pro XML, Arcade only).dat" --input "arcade/*.zip" --output 1G1R/ --only-retail --single --prefer-region USA,WORLD,EUR,JPN --prefer-language EN --prefer-revision newer --prefer-parent
`

I tested a few games and it seems it works. 

I'd rather have the BIOS files outside the roms, since RetroArch handles them... did I miss this?

**Reply by emmercm** [reply]
I think arcade DATs typically use the first release as the parent, so it shouldn't be a bootleg or hack, but I opened https://github.com/emmercm/igir/issues/1349 just in case.

For arcade, you probably don't want `--prefer-region` or `--prefer-language`, arcade DATs don't tend to have this information available like non-arcade DATs do.

Arcade BIOSes are also different from non-arcade BIOSes in that you don't typically put them in the emulator's BIOS directory ("system" in the case of RetroArch). Arcade emulators expect the BIOS files to live alongside the ROMs, with location depending on the [merge mode](https://igir.io/usage/arcade/#rom-set-merge-types). You'll want to keep them included, and it's a good idea to choose a merge mode explicitly.

Let me know if there's anything else I can help with!

**Comment by notNSANE** [comment]
Made new batch:

```
@echo off

igir.exe copy clean ^
 --dat "FinalBurn Neo (ClrMame Pro XML, Arcade only).dat" ^
 --input "./arcade/*.zip" ^
 --output "./1G1R/" ^
 --only-retail ^
 --single ^
 --prefer-revision newer ^
 --prefer-parent ^
 --merge-roms fullnonmerged

pause
```

Even after the changes, I got the exact same number of files and file size, so I guess this is the best it can do.

---

Besides all that, I get a ton of these errors: 

```
WARN:  Arcade Games: no single archive contains all necessary files, cannot copy these different input files to:
1G1R\zteamdw.zip
```

I guess they're safe to skip?

**Reply by notNSANE** [reply]
ok, I don't have time right now to find every detail, but found some "undesirables"

* first, bioses as separate files, I added --no-bios to my bat.
* Captain America And The Avengers (Asia Rev 1.4) (captaven.zip). I guess the revision command took over? this needs documentation but the world version should have priority.
* Final Fight (ffightaec2.zip). It's a romhack, it shouldn't be on this list... but it's a good romhack, lol. Here's more info: https://github.com/originalgrego/FinalFightAE-Source

**Reply by notNSANE** [reply]
found a huge issue: all metal slug (mslug, mslug2, etc.) games except metal slug 5 (mslug5pcb) were removed. What happened here?

**Reply by notNSANE** [reply]
sorry to annoy again @emmercm , any idea what happened with Metal Slug? Because it most probably happened with a lot of other valid games.

**Reply by hlarsen** [reply]
> sorry to annoy again @emmercm , any idea what happened with Metal Slug? Because it most probably happened with a lot of other valid games.

I'm using `copy zip --no-unlicensed --only-retail --single --prefer-revision newer` with separate input and output directories and i can't reproduce your issue with metal slug - I end up with 7 `mslug*.zip` files rather than the 14 in the full mame set i used as input.

> Captain America And The Avengers (Asia Rev 1.4) (captaven.zip). I guess the revision command took over? this needs documentation but the world version should have priority.

This one i can reproduce. In the official MAME DAT, `captavenu` is US v1.9 but it's set as a clone of `captaven`, which is Asia v1.4. Using the same `copy zip` command as above I've tried with and without `--prefer-parent` and `--prefer-revision newer` and I get `captaven` each time. The below logs show it being selected so I'm guessing igir isn't trying to parse this stuff:

```
[19:31:28.689] TRACE: CandidatePreferer: 0.271 (mame0271): captaven (parent): 7 candidates before filtering                                                                                                                                                    
[19:31:28.689] TRACE: CandidatePreferer: 0.271 (mame0271): captaven: preferred captaven                                                                                                                                                                        
```
